src/modules/idyntree-yarp-visualizer/RobotConnectors.cpp

```diff
@@ -0,0 +1,558 @@
+        }
+
+        yarp::os::Bottle* controlBoardsList = controlBoardsValue.asList();
+
+        for (size_t i = 0; i < controlBoardsList->size(); ++i)
+        {
+            yarp::os::Value cbValue = controlBoardsList->get(i);
+            if (!cbValue.isString())
+            {
+                yError() << "The value in position" << i << "(0-based) of controlBoards is not a string.";
+                return false;
+            }
+            m_controlBoards.push_back(cbValue.asString());
+        }
+    }
+
+    return true;
+}
+
+bool RemapperConnector::getJointNamesFromModel(const yarp::os::Searchable &inputConf, const iDynTree::Model &model)
+{
+    std::lock_guard<std::mutex> lock(m_basicInfo->mutex);
+
+    m_basicInfo->jointList.clear();
+
+    yarp::os::Value jointsValue = inputConf.find("joints");
+
+    if (jointsValue.isNull())
+    {
+        for (size_t i = 0; i < model.getNrOfJoints(); ++i)
+        {
+            if (model.getJoint(i)->getNrOfDOFs() == 1)
+            {
+                m_basicInfo->jointList.push_back(model.getJointName(i)); //automatically select all the joint in the model that have 1DOF
+            }
+        }
+        if (m_basicInfo->jointList.size() != model.getNrOfPosCoords())
+        {
+            yError() << "The model contains joints with more than 1DOF. This is not yet supported.";
+            return false;
+        }
+    }
+    else
+    {
+        if (!jointsValue.isList())
+        {
+            yError() << "joints is specified, but it is not a list.";
+            return false;
+        }
+
+        yarp::os::Bottle* jointList = jointsValue.asList();
+
+        for (size_t i = 0; i < jointList->size(); ++i)
+        {
+            yarp::os::Value jvalue = jointList->get(i);
+            if (!jvalue.isString())
+            {
+                yError() << "The value in position" << i << "(0-based) of joints is not a string.";
+                return false;
+            }
+            m_basicInfo->jointList.push_back(jvalue.asString());
+        }
+    }
+
+    m_jointsInDeg.resize(m_basicInfo->jointList.size());
+    m_jointsInDeg.zero();
+
+    return true;
+}
+
+bool RemapperConnector::configure(const yarp::os::Searchable &inputConf, const iDynTree::Model &fullModel, std::shared_ptr<BasicInfo> basicInfo)
+{
+    std::lock_guard<std::mutex> lock(m_mutex);
+
+    m_basicInfo = basicInfo;
+
+    if (!getOrGuessControlBoardsFromFile(inputConf))
+    {
+        yError() << "Failed to get the controlboards list.";
+        return false;
+    }
+
+    if (!getJointNamesFromModel(inputConf, fullModel))
+    {
+        yError() << "Failed to get the list of joints.";
+        return false;
+    }
+
+    return true;
+}
+
+bool RemapperConnector::connectToRobot()
+{
+    m_connected = false;
+
+    std::lock_guard<std::mutex> lock(m_mutex);
+    {
+        std::lock_guard<std::mutex> lock(m_basicInfo->mutex);
+        m_robotDevice.close();
+        m_encodersInterface = nullptr;
```

Comment:
a comment here saying resetting for safety would be helpful for future reference?

Reply:
I am not sure. Those lines are there to make sure that the method can be called multiple times without crashing. This is helpful in case the robot is rebooted for example. On the other hand, printing a message like "Resetting connection" every time might be confusing.

Reply:
Didn't mean for it to be a debug print, just a comment so that when another reader comes back to the code, they might understand why this is being done here. This is extremely minor (feel free to ignore as well).

Reply:
I will add it in case of other commits!

Reply:
Done in https://github.com/robotology/idyntree-yarp-tools/pull/7/commits/898eabd9a5d783fc6a0110d3fc3476d6994c3296